File: Prácticas/P4/RUBY/Civitas/lib/casilla_impuesto.rb
# To change this license header, choose License Headers in Project Properties.
# To change this template file, choose Tools | Templates
# and open the template in the editor.

module Civitas
  class CasillaImpuesto < Casilla
    def initialize(cantidad, nombre)
      @importe = cantidad
      super(nombre)
    end
    
    def recibe_jugador(actual,todos)
      if(jugador_correcto(actual,todos))
        informe(actual, todos)
        todos[actual].paga_impuestos(@importe)
      end
    end
    
    def to_s
      to_s="CasillaImpuesto importe= #{@importe}"
    end
  end
end

File: Prácticas/P4/JAVA/Civitas/src/civitas/CasillaImpuesto.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package civitas;

import java.util.ArrayList;

/**
 *
 * @author Jose Padial
 */
public class CasillaImpuesto extends Casilla{
    private float importe;
    
    CasillaImpuesto(String nombre, float cantidad){
        super(nombre);
        this.importe = cantidad;
    }
    
    @Override
    void recibeJugador(int actual, ArrayList<Jugador> todos){
        if(jugadorCorrecto(actual,todos) == true){
            informe(actual,todos);
            todos.get(actual).pagaImpuesto(importe);
        }
    }

    @Override
    public String toString() {
        return "CasillaImpuesto{" + "importe=" + importe + '}';
    }
}
